### PR TITLE
Doxygen2: Electric Boogaloo

### PR DESCRIPTION
**Pull request by jcantrell**
Now with 50% less extraneous changes!

(Just adding a doxygen configuration file)

### CODE REVIEW

File: doxygen.config [hunk 1]
```diff
@@ -0,0 +1,2427 @@
...
+
+DOXYFILE_ENCODING      = UTF-8
+
+# The PROJECT_NAME tag is a single word (or a sequence of words surrounded by
+# double-quotes, unless you are using Doxywizard) that should identify the
+# project for which the documentation is generated. This name is used in the
+# title of most generated pages and in a few other places.
+# The default value is: My Project.
+
+PROJECT_NAME           = "My Project"
```
Comment (connorkuehl) [inline doxygen.config]: Clang Randstruct

File: doxygen.config [hunk 2]
```diff
@@ -0,0 +1,2427 @@
...
+# could be handy for archiving the generated documentation or if some version
+# control system is used.
+
+PROJECT_NUMBER         =
+
+# Using the PROJECT_BRIEF tag one can provide an optional one line description
+# for a project that appears at the top of each page and should give viewer a
+# quick idea about the purpose of the project. Keep the description short.
+
+PROJECT_BRIEF          =
```
Comment (connorkuehl) [inline doxygen.config]: Something like "Clang Randstruct is a compiler plugin that enables in-memory structure layout randomization for compile-time hardening"

File: doxygen.config (hunk 1 above)
Comment (jcantrell) [reply doxygen.config]: done

File: doxygen.config (hunk 2 above)
Comment (jcantrell) [reply doxygen.config]: & done

**Comment by connorkuehl** [comment]
It is a masterpiece!

**Pull request merged**